app/uk/gov/hmrc/tai/connectors/CacheConnector.scala

```diff
@@ -94,138 +79,97 @@ class CacheConnector @Inject()(
     } else {
       Json.toJson(data)
     }
-    cacheRepository.createOrUpdate(cacheId.value, key, jsonData).map(_ => data)
+    taiCacheRepository.save(cacheId.value)(key, jsonData).map(_ => data)
```

Comment:
not sure this is worth having a separate method for. We could just use the MongoCacheRepository put method and do
`taiCacheRepository.save(cacheId.value)(DataKey(key), jsonData)`

Reply:
The 'put' was causing issues in Mockito so had to wrap it here to facilitate testing